scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -231,49 +232,58 @@ def _str2bool(v: str) -> bool:
     logger.print(f"create list: {publish_list}")
 
     failure_list = []
-    for asset in util.find_assets(input_dirs=assets_dir):
-        asset_names = publish_list.get(asset.type.value, [])
-        if not ("*" in asset_names or asset.name in asset_names):
-            logger.print(
-                f"Skipping registering asset {asset.name} because it is not in the publish list")
+    all_assets = util.find_assets(input_dirs=assets_dir)
+    assets_by_type = {}
+    for asset in all_assets:
+        assets_by_type[asset.type.value] = assets_by_type.get(asset.type.value, []).append(asset)
```

Comment:
defaultdict is perfect here. You'll need to add `from collections import defaultdict` up above.
```suggestion
    assets_by_type = defaultdict(list)
    for asset in all_assets:
        assets_by_type[asset.type].append(asset)
```